### PR TITLE
fix: Small static analysis findings

### PR DESCRIPTION
**Pull request by timtebeek**
### Context
Hi! Saw you recently added Sonar and wanted to reduce the number of findings; figured I'd see if I could help with a few automated fixes; hope it's appreciated.

I chose to run only a select few recipes, as these had a manageable changeset and hopefully desirable outcomes. Here's a log of what changes were made where.

```
Changes have been made to plugins/jdks-maven-plugin/build/mavenPlugin/helpMojo/org/jreleaser/jdks/maven/plugin/help/HelpMojo.java by:
    org.openrewrite.java.cleanup.UseDiamondOperator
Changes have been made to sdks/jreleaser-codeberg-java-sdk/src/main/java/org/jreleaser/sdk/codeberg/CodebergReleaser.java by:
    org.openrewrite.java.cleanup.EqualsAvoidsNull
Changes have been made to core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JpackageAssemblerProcessor.java by:
    org.openrewrite.java.cleanup.EqualsAvoidsNull
Changes have been made to sdks/jreleaser-git-java-sdk/src/main/java/org/jreleaser/sdk/git/ChangelogGenerator.java by:
    org.openrewrite.java.cleanup.EqualsAvoidsNull
Changes have been made to sdks/jreleaser-gitea-java-sdk/src/main/java/org/jreleaser/sdk/gitea/GiteaReleaser.java by:
    org.openrewrite.java.cleanup.EqualsAvoidsNull
Changes have been made to sdks/jreleaser-gitlab-java-sdk/src/main/java/org/jreleaser/sdk/gitlab/GitlabReleaser.java by:
    org.openrewrite.java.cleanup.EqualsAvoidsNull
Changes have been made to plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/internal/KordampJReleaserAdapter.groovy by:
    org.openrewrite.java.cleanup.IsEmptyCallOnCollections
Changes have been made to api/jreleaser-logger-api/src/main/java/org/jreleaser/logging/AbstractJReleaserLogger.java by:
    org.openrewrite.java.cleanup.EqualsAvoidsNull
Changes have been made to plugins/jreleaser-maven-plugin/build/mavenPlugin/helpMojo/org/jreleaser/maven/plugin/help/HelpMojo.java by:
    org.openrewrite.java.cleanup.UseDiamondOperator
Changes have been made to api/jreleaser-model-api/src/main/java/org/jreleaser/version/SemanticVersion.java by:
    org.openrewrite.java.cleanup.EqualsAvoidsNull
Changes have been made to api/jreleaser-utils/src/main/java/org/jreleaser/util/CollectionUtils.java by:
    org.openrewrite.java.cleanup.UseDiamondOperator
Changes have been made to api/jreleaser-utils/src/test/java/org/jreleaser/util/CollectionUtilsTest.java by:
    org.openrewrite.java.cleanup.UseDiamondOperator
Changes have been made to sdks/jreleaser-webhooks-java-sdk/src/main/java/org/jreleaser/sdk/webhooks/WebhooksAnnouncer.java by:
    org.openrewrite.java.cleanup.EqualsAvoidsNull
```

If needed I can recreate these changes excluding for instance the `EqualsAvoidsNull` if that does not fit the desired code style.

As these changes were mostly automated I'm not particularly attached to them, so if you feel it does not improve the project feel free to not accept the contribution; no harmed feelings. :)

### Checklist
- [x] [Review Contribution Guidelines](https://github.com/jreleaser/jreleaser/blob/master/CONTRIBUTING.adoc).
- [x] Make sure all contributed code can be distributed under the terms of the 
      [Apache License 2.0](https://github.com/jreleaser/jreleaser/blob/master/LICENSE), e.g. the code was written by 
      you or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Update [documentation](https://github.com/jreleaser/jreleaser.github.io) when applicable.

### CODE REVIEW

File: core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JpackageAssemblerProcessor.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ private void adjustRuntimeImage(JReleaserContext context, JpackageAssembler asse
 
         try {
             if (!FileUtils.copyFilesRecursive(context.getLogger(), originalImage, adjustedImage, path -> {
-                boolean pathIsJar = path.getFileName().toString().endsWith(".jar") && path.getParent().getFileName().toString().equals("jars");
+                boolean pathIsJar = path.getFileName().toString().endsWith(".jar") && "jars".equals(path.getParent().getFileName().toString());
```
Comment (aalmiray) [inline core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JpackageAssemblerProcessor.java:144]: This conditional could still throw NPEs if an intermediate result in the chain `path.getParent().getFileName().toString()` were to be null. However, it won't. I think this change may be omitted.

Comment (timtebeek) [reply core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JpackageAssemblerProcessor.java:144]:
```suggestion
                boolean pathIsJar = path.getFileName().toString().endsWith(".jar") && path.getParent().getFileName().toString().equals("jars");
```

Comment (timtebeek) [reply core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JpackageAssemblerProcessor.java]: Thought I could do this quickly from GitHub, but forgot about the preference for squashed & commit message conventions; I'll force push! :)

**Comment by aalmiray** [comment]
Thanks! 😄

File: core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JpackageAssemblerProcessor.java (hunk 1 above)
Comment (aalmiray) [reply core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JpackageAssemblerProcessor.java]: It's OK. I've squashed on my side 😏

**Comment by timtebeek** [comment]
Happy to help & hope to see you at some events again this summer. :)

**Comment by aalmiray** [comment]
Released in v1.5.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.5.0

**Pull request merged**